### PR TITLE
[Flask Server] A bunch of improvement clean up to get read for integration test

### PR DESCRIPTION
**Pull request by shifucun**
* Use absolute path for configs
* Add pytest config file so it's easier to see log output
* Move all flask app code inside create_app() function
* extract backend check into util functions
* disable "ADMIN" in configmodule as that is not used in any instances
* Only use "POST" for /nlnext/data as the code does not work for "GET"

### CODE REVIEW

File: server/lib/util.py [hunk 1]
```diff
@@ -41,11 +45,14 @@
     'poverty': ['USA', 'India'],
 }
 
+_root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
Comment (pradh) [inline server/lib/util.py:48]: should we put this in a more general util and use it?

**Review by pradh — COMMENTED** [review]
Nice, LG!  @beets is probably a better reviewer for the changes.

File: server/lib/util.py (hunk 1 above)
Comment (shifucun) [reply server/lib/util.py]: Good idea. Made a function, but still can not share with nl_server for now.

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[pytest]
+log_cli=true
+log_level=INFO
+log_file=log.txt
```
Comment (beets) [inline pytest.ini:4]: maybe pytest_log.txt?

File: server/lib/util.py [hunk 3]
```diff
@@ -196,4 +208,37 @@ def _compact_series(series_resp, all_facets):
...
+    # Codacity still calls out the error so disable the check.
+    # https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b310-urllib-urlopen
+    urllib.request.urlopen(url)  # nosec B310
+    return True
+  except urllib.error.URLError:
+    return False
+
+
+def check_backend_ready(urls: List[str]):
+  timeout = 120  # seconds
```
Comment (beets) [inline server/lib/util.py:229]: use a constant for 120?

**Review by beets — APPROVED** [review]

File: server/lib/util.py (hunk 3 above)
Comment (shifucun) [reply server/lib/util.py:229]: Done

File: pytest.ini (hunk 2 above)
Comment (shifucun) [reply pytest.ini:4]: Done

**Review by shifucun — COMMENTED** [review]
Thanks for review!

**Pull request merged**